src/main.vala

```diff
@@ -88,7 +88,23 @@ class Vls.Server : Object {
     delegate void CallHandler (Vls.Server self, Jsonrpc.Server server, Jsonrpc.Client client, string method, Variant id, Variant @params);
 
     private void log_handler (string? log_domain, LogLevelFlags log_levels, string message) {
-        printerr ("%s: %s\n", log_domain == null ? "vls" : log_domain, message);
+        string level = "";
+
+        if ((log_levels & LogLevelFlags.LEVEL_MASK) == LogLevelFlags.LEVEL_MASK)
+            level = "-ALL";
+        else if ((log_levels & LogLevelFlags.LEVEL_CRITICAL) != 0)
+            level = "-CRITICAL";
+        else if ((log_levels & LogLevelFlags.LEVEL_DEBUG) != 0)
+            level = "-DEBUG";
+        else if ((log_levels & LogLevelFlags.LEVEL_ERROR) != 0)
+            level = "-ERROR";
+        else if ((log_levels & LogLevelFlags.LEVEL_INFO) != 0)
+            level = "-INFO";
+        else if ((log_levels & LogLevelFlags.LEVEL_MESSAGE) != 0)
+            level = "-MESSAGE";
+        else if ((log_levels & LogLevelFlags.LEVEL_WARNING) != 0)
+            level = "-WARNING";
+        printerr ("%s: %s\n", log_domain == null ? @"vls$level" : log_domain, message);
```

Comment:
doesn't the default glib log handler do this already?

Reply:
Except this prints everything to stderr (`printerr(...)`).